tlsfuzzer/expect.py

```diff
@@ -433,7 +463,53 @@ def gen_srv_ext_handler_record_limit(size=None):
          ExtensionType.record_size_limit: _srv_ext_handler_record_limit}
 
 
-class ExpectServerHello(ExpectHandshake):
+_CR_EXT_HANDLER = \
+        {ExtensionType.status_request: clnt_ext_handler_status_request,
+         ExtensionType.signature_algorithms: clnt_ext_handler_sig_algs,
+         ExtensionType.signature_algorithms_cert: clnt_ext_handler_sig_algs}
+
+
+class _ExpectExtensionsMessage(ExpectHandshake):
+    """
+    Common methods of messages that have a list of extensions.
+
+    Used in ServerHello, EncryptedExtensions and CertificateRequest (in
+    TLS 1.3)
+    """
+    def __init__(self, content_type, msg_type, extensions):
+        super(_ExpectExtensionsMessage, self).__init__(
+            content_type, msg_type)
+        self.extensions = extensions
+
+    def _compare_extensions(self, message):
+        """
+        Verify that server provided extensions match exactly expected list.
+        """
+        # if the list of extensions is present, make sure it matches exactly
+        # with what the server sent
+        if self.extensions and not message.extensions:
+            raise AssertionError("Server did not send any extensions")
+        if self.extensions is not None and message.extensions:
```

Comment:
Did you want to detect if you weren't expecting extensions, but found some (self.extensions is None and message.extensions)?

Reply:
Never mind. It looks like that's handled in _compare_extensions_in_ee. Probably should say so in the comments.

Reply:
`None` for value is the general idiom of "accept anything, as long as allowed by the protocol"
it is also spelled out in the doc string for both ExpectServerHello and ExpectCertificateRequest